### ISSUE TITLE
account.Address chokes on malformed address headers

### ISSUE THREAD

**Issue by pazz** [pazz/alot#1152]
This happens on the master branch.
I received a spam mail `From: "Booking.com" <Booking.com>`. Note the malformed address.
Trying to display this message in search or thread mode blows up the new Address class:
```
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/defer.py", line 150, in maybeDeferred
    result = f(*args, **kw)
  File "/home/pazz/projects/alot/alot/commands/globals.py", line 126, in apply
    sort_order=self.order))
  File "/home/pazz/projects/alot/alot/ui.py", line 375, in buffer_open
    self.buffer_focus(buf)
  File "/home/pazz/projects/alot/alot/ui.py", line 435, in buffer_focus
    self.update()
  File "/home/pazz/projects/alot/alot/ui.py", line 621, in update
    self.mainloop.draw_screen()
  File "/usr/lib/python2.7/dist-packages/urwid/main_loop.py", line 578, in draw_screen
    canvas = self._topmost_widget.render(self.screen_size, focus=True)
  File "/usr/lib/python2.7/dist-packages/urwid/widget.py", line 141, in cached_render
    canv = fn(self, size, focus=focus)
  File "/usr/lib/python2.7/dist-packages/urwid/decoration.py", line 225, in render
    canv = self._original_widget.render(size, focus=focus)
  File "/usr/lib/python2.7/dist-packages/urwid/widget.py", line 141, in cached_render
    canv = fn(self, size, focus=focus)
  File "/usr/lib/python2.7/dist-packages/urwid/container.py", line 1083, in render
    focus and self.focus_part == 'body')
  File "/home/pazz/projects/alot/alot/buffers.py", line 39, in render
    return self.body.render(size, focus)
  File "/usr/lib/python2.7/dist-packages/urwid/widget.py", line 141, in cached_render
    canv = fn(self, size, focus=focus)
  File "/usr/lib/python2.7/dist-packages/urwid/listbox.py", line 457, in render
    (maxcol, maxrow), focus=focus)
  File "/usr/lib/python2.7/dist-packages/urwid/listbox.py", line 339, in calculate_visible
    self._set_focus_complete( (maxcol, maxrow), focus )
  File "/usr/lib/python2.7/dist-packages/urwid/listbox.py", line 704, in _set_focus_complete
    (maxcol,maxrow), focus)
  File "/usr/lib/python2.7/dist-packages/urwid/listbox.py", line 674, in _set_focus_first_selectable
    (maxcol, maxrow), focus=focus)
  File "/usr/lib/python2.7/dist-packages/urwid/listbox.py", line 342, in calculate_visible
    focus_widget, focus_pos = self.body.get_focus()
  File "/home/pazz/projects/alot/alot/walker.py", line 31, in get_focus
    return self._get_at_pos(self.focus)
  File "/home/pazz/projects/alot/alot/walker.py", line 64, in _get_at_pos
    widget = self._get_next_item()
  File "/home/pazz/projects/alot/alot/walker.py", line 77, in _get_next_item
    next_widget = self.containerclass(next_obj, **self.kwargs)
  File "/home/pazz/projects/alot/alot/widgets/search.py", line 28, in __init__
    self.rebuild()
  File "/home/pazz/projects/alot/alot/widgets/search.py", line 149, in rebuild
    minw, maxw, align_mode)
  File "/home/pazz/projects/alot/alot/widgets/search.py", line 76, in _build_part
    authors = self.thread.get_authors_string() or '(None)'
  File "/home/pazz/projects/alot/alot/db/thread.py", line 198, in get_authors_string
    if aaddress in own_addrs:
  File "/home/pazz/projects/alot/alot/account.py", line 144, in __eq__
    return self.__cmp(other, operator.eq)
  File "/home/pazz/projects/alot/alot/account.py", line 125, in __cmp
    ouser, odomain = other.split(u'@')
exceptions.ValueError: need more than 1 value to unpack
```

I suspect that #1150 is related.

**Comment by dcbaker** [comment]
#1151 fixes this

I'm on mobile, please excuse autocorrect fail.

**Comment by pazz** [comment]
fixed through #1151

**Comment by pazz** [comment]
thanks @dcbaker !
